packages/components/table/src/util.ts

```diff
@@ -120,6 +120,8 @@ export const getColumnByKey = function <T>(
       break
     }
   }
+  if (!column)
+    throw new Error(`[Table] No column matching with column-key: ${columnKey}`)
```

Comment:
According to @btea's suggestion, it's better to use `throwError`. https://github.com/element-plus/element-plus/pull/9322#discussion_r956541365

```ts
import {  throwError } from '@element-plus/utils'
```

```ts
throwError('ElTable', `No column matching with column-key: ${columnKey}`)
```